### PR TITLE
[REMOVAL] Drop TYPO3 7.6 LTS support

### PR DESCRIPTION
**Pull request by NamelessCoder**

### CODE REVIEW

**Comment by cedricziel** [comment]
Same thing. Nice! I hereby propose we drop the hell out of the 7.6 support for the development branch.

In doing so we could drop the `typo3/cms` dependency on all fronts and could go back to a stable dependency graph. WDYT?

**Comment by NamelessCoder** [comment]
Question, @cedricziel - can you spot in our dependency tree what the deuce is causing `typo3/cms` to be selected in the `dev-master` build when it properly selects explicits in all others? I've looked and looked and can't find any traces why that package is attempted...

**Comment by cedricziel** [comment]
Let's have look!

**Comment by NamelessCoder** [comment]
btw, right after this one I expect to make the v9 compat one, just didn't want to mix it up with the removal.

**Comment by cedricziel** [comment]
Found it. The error message is wrong - in the current version of the composer plugin for typo3, the install of `typo3/cms-core` is blocked, not `typo3/cms`.

**Comment by cedricziel** [comment]
https://github.com/TYPO3/CmsComposerInstallers/blob/master/src/Installer/Plugin.php#L72

in combination with

https://github.com/TYPO3/CmsComposerInstallers/blob/master/src/Installer/CoreInstaller.php#L39

**Comment by NamelessCoder** [comment]
Interesting! Nice find :)

File: .travis.yml [hunk 1]
```diff
@@ -31,7 +27,11 @@ before_script:
   - export TYPO3_PATH_WEB=$PWD/build/web
 
 install:
-  - composer require --no-update "typo3/cms:$TYPO3_VERSION"
+  - composer require --no-update "typo3/cms-core:$TYPO3_VERSION"
```
Comment (cedricziel) [inline .travis.yml]: Please remove this line

File: composer.json [hunk 2]
```diff
@@ -32,14 +33,18 @@
     },
     "require": {
         "php": ">=7.0.0",
-        "typo3/cms": ">=7.6.13, <=7.6.99|^8.4.1|dev-master"
+        "typo3/cms-core": "^8.7|dev-master",
```
Comment (cedricziel) [inline composer.json]: Please remove this line.

File: .travis.yml (hunk 1 above)
Comment (NamelessCoder) [reply .travis.yml]: Hmm, then it won't update EXT:core when running the tests - or?

**Comment by NamelessCoder** [comment]
Something strange is afoot here. I've now tried this `composer.json` require section:

```
   "require": {
        "php": ">=7.0.0",
        "typo3/cms-core": "^9.1",
        "typo3/cms-extbase": "^9.1",
        "typo3/cms-fluid": "^9.1",
        "typo3/cms-frontend": "^9.1",
        "typo3/cms-backend": "^9.1"
    },
```

And doing `composer install --no-dev` yields:

```
macbook:vhs claus$ composer install --no-dev
Loading composer repositories with package information
Updating dependencies
Package operations: 31 installs, 0 updates, 0 removals
  - Installing typo3/cms-composer-installers (dev-master f7b959d): Loading from cache
  - Installing typo3/class-alias-loader (dev-master d067bad): Loading from cache
  - Installing typo3/cms-cli (2.0.0): Loading from cache
  - Installing mso/idna-convert (v1.1.0): Loading from cache
  - Installing cogpowered/finediff (0.3.1): Loading from cache
  - Installing symfony/yaml (dev-master 90fbbce): Loading from cache
  - Installing symfony/polyfill-mbstring (dev-master 78be803): Loading from cache
  - Installing psr/log (dev-master 4ebe3a8): Loading from cache
  - Installing symfony/console (dev-master 524258a): Loading from cache
  - Installing psr/http-message (dev-master f6561bf): Loading from cache
  - Installing guzzlehttp/psr7 (dev-master 649e354): Loading from cache
  - Installing guzzlehttp/promises (dev-master 0940c91): Loading from cache
  - Installing guzzlehttp/guzzle (dev-master ba1bce3): Loading from cache
  - Installing doctrine/instantiator (1.0.x-dev 8e884e7): Loading from cache
  - Installing symfony/intl (dev-master 3a45a76): Loading from cache
  - Installing symfony/polyfill-intl-icu (dev-master 254919c): Loading from cache
  - Installing nikic/php-parser (3.x-dev bb87e28): Loading from cache
  - Installing doctrine/lexer (dev-master cc709ba): Loading from cache
  - Installing doctrine/inflector (dev-master c5313c1): Loading from cache
  - Installing doctrine/collections (dev-master f489500): Loading from cache
  - Installing doctrine/cache (dev-master e3fcea0): Loading from cache
  - Installing doctrine/annotations (dev-master 9419fd5): Loading from cache
  - Installing typo3fluid/fluid (2.5.3): Loading from cache
  - Installing psr/container (dev-master 2cc4a01): Loading from cache
  - Installing doctrine/common (dev-master a61bfdd): Loading from cache
  - Installing doctrine/dbal (dev-master 3bfaa82): Loading from cache
  - Installing psr/http-server-handler (dev-master ede6c50): Loading from cache
  - Installing psr/http-server-middleware (dev-master e012371): Loading from cache
  - Installing symfony/finder (dev-master 6347bc2): Loading from cache
  - Installing swiftmailer/swiftmailer (v5.4.9): Loading from cache
Installation of typo3/cms is not possible any more for TYPO3 versions 9.0.0 and higher.
Please require typo3/minimal instead for minimum required TYPO3 system extensions,
and/or require individual system extensions like typo3/cms-extension-name
E.g. composer require typo3/cms-tstemplate
```

**Comment by cedricziel** [comment]
Please remove the `typo3/cms-core` dependency as mentioned above

**Comment by cedricziel** [comment]
I proposed a change to the composer plugin to reflect the package type constraint (https://github.com/TYPO3/CmsComposerInstallers/pull/76)

**Comment by NamelessCoder** [comment]
The error is the same without the explicit core dep...

**Comment by cedricziel** [comment]
Push it, push it

**Comment by NamelessCoder** [comment]
I'm beginning to suspect that if we want to continue testing on TYPO3 dev-master, we're going to need a special section in Travis where we strip all TYPO3 deps and add typo3/cms:dev-master as the only dep...

**Comment by NamelessCoder** [comment]
I've restored the deps that work if you install any other than `dev-master` of TYPO3. We can take care of testing on `dev-master` another day.

**Comment by cedricziel** [comment]
Yes we can.

**Comment by cedricziel** [comment]
Thx for your time!

**Comment by NamelessCoder** [comment]
You too :)

**Comment by helhum** [comment]
> The error message is wrong - in the current version of the composer plugin for typo3, the install of typo3/cms-core

That is a wrong assumption. The package `typo3/cms-core` is of type `typo3-cms-framework`, not of type `typo3-cms-core`.

When requiring `typo3/cms-composer-installers`, it was never possible that any other package of type `typo3-cms-core` existed

**Comment by cedricziel** [comment]
@helhum Yep, that's what i understood in the meantime - thank you for your time. The proposed change is still valid.

**Comment by helhum** [comment]
The issue you are facing with the master build is not, that typo3/cms is in the dependency tree, but that `typo3/cms-version` is required by `fluidtypo3/development`. This package does not exist, except as replacement for `typo3/cms` (it was removed completely in TYPO3 8.7)

There are a couple of solutions for it:

1. Remove `typo3/cms-version` dependency (why is it needed in this package?)
1. Create a change request for TYPO3, that adds `typo3/cms-version` as replace for `typo3/cms-workspaces`
1. Add `typo3/cms-version` as replace in this package (as it isn't compatible with TYPO3 7.6 any more anyway)

Suggested change: #1458

**Pull request merged**